config/samples/beats/3_metricbeat-kubernetes.yaml

```diff
@@ -0,0 +1,383 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: metricbeat-daemonset-config
+  namespace: beats
+  labels:
+    k8s-app: metricbeat
+data:
+  metricbeat.yml: |-
+    metricbeat.config.modules:
+      # Mounted `metricbeat-daemonset-modules` configmap:
+      path: ${path.config}/modules.d/*.yml
+      # Reload module configs as they change:
+      reload.enabled: false
+
+    # To enable hints based autodiscover uncomment this:
+    metricbeat.autodiscover:
+      providers:
+        - type: kubernetes
+          host: ${NODE_NAME}
+          hints.enabled: true
+
+    setup.dashboards.enabled: true
```

Comment:
Hey! How about moving all these kibana related stuff to the Deployment config map?

Deployment stands as a singleton for the whole cluster so it will far less expensive to load the dashboards from there instead of from each of the Daemonset instances.

Also maybe using env vars for the `KIBANA_HOST` and `KIBANA_PORT` might make it prettier.

See also: https://github.com/elastic/beats/pull/14131/files#diff-6c4e791fd61d7fd51348955d49d87917R219

Reply:
I followed your examples. I am not sure about where the right place for these examples is, feels wrong to duplicate it in both repos? 

I also raised https://github.com/elastic/cloud-on-k8s/issues/2051 to discuss next steps